### PR TITLE
exclude the cordoned nodes when calc node allocatable

### PR DESCRIPTION
**Pull request by skeeey**
Signed-off-by: liuwei <liuweixa@redhat.com>

### CODE REVIEW

**Comment by skeeey** [comment]
refer to https://github.com/open-cluster-management-io/community/issues/77

File: pkg/spoke/managedcluster/node_controller.go [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+}
+
+func (c *managedClusterNodeController) sync(ctx context.Context, syncCtx factory.SyncContext) error {
+	managedCluster, err := c.hubClusterLister.Get(c.clusterName)
+	if err != nil {
+		return fmt.Errorf("unable to get managed cluster with name %q from hub: %w", c.clusterName, err)
+	}
+
+	// current managed cluster is not accepted, do nothing.
+	if !meta.IsStatusConditionTrue(managedCluster.Status.Conditions, clusterv1.ManagedClusterConditionHubAccepted) {
```
Comment (qiujian16) [inline pkg/spoke/managedcluster/node_controller.go]: should this be Available?

**Review by qiujian16 — COMMENTED** [review]
why do we separate this  to another controller? It will double the update call, right?

File: pkg/spoke/managedcluster/node_controller.go (hunk 1 above)
Comment (skeeey) [reply pkg/spoke/managedcluster/node_controller.go]: yes, the `Available` is more suitable than `Accepted`

**Comment by skeeey** [comment]
> why do we separate this to another controller? It will double the update call, right?

I think separate this make the code logic more clear, currently, we maintain this in the cluster joining controller, but collecting the node resource and handling cluster join event may be two irrelevant tasks

when the cluster starts to register, this will add an update call, but if we use  cluster `Available` as the judgement condition, I think we can stagger this update from others

what's your opinion? do you think we should keep this in the joining controller?

**Comment by qiujian16** [comment]
I think we can move it as part of health check (available) controller

**Comment by skeeey** [comment]
agree with you, and I think we can also put the get version in health check controller, thus the joining controller will only handle the cluster joining

**Pull request merged**